packages/desktop-client/src/components/accounts/TransactionsTable.js

```diff
@@ -1787,6 +1837,120 @@ export let TransactionTable = forwardRef((props, ref) => {
     id => splitsExpanded.dispatch({ type: 'toggle-split', id }),
     [splitsExpanded.dispatch],
   );
+  let [field, setField] = useState(null);
+  let [ascDesc, setAscDesc] = useState('desc');
+
+  function onSortTable(e) {
+    let transform =
+      e.currentTarget.innerText.toLowerCase() === 'payment' ||
+      e.currentTarget.innerText.toLowerCase() === 'deposit'
+        ? 'amount'
+        : e.currentTarget.innerText.toLowerCase();
+    if (transform === field) {
+      ascDesc === 'desc' ? setAscDesc('asc') : setAscDesc('desc');
+    } else {
+      setField(transform);
+      transform === 'amount' || transform === 'date'
+        ? setAscDesc('asc')
+        : setAscDesc('desc');
+    }
+  }
+
+  let category = props.categoryGroups.flatMap(cat => {
+    return cat.categories;
+  });
+  let fields = [field] + 's';
+
+  function sortStrings() {
+    transactions =
+      ascDesc === 'asc'
+        ? transactions
+            .map(e => {
+              return {
+                ...e,
+                test: props[fields].find(f => {
+                  return f.id === e[field];
+                })
+                  ? props[fields].find(f => {
+                      return f.id === e[field];
+                    }).name
+                  : '',
+              };
+            })
```

Comment:
This bit seems to be pretty repetitive — can you pull it out? Also, you could simplify it a bit by doing `props[fields].find(f => f.id === e[field])?.name ?? ''`

Reply:
Thank you, I knew there was a better way to do it. You're a wizard!